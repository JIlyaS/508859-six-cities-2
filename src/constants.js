export const RATING_PERSENT = 100;
export const MAX_RATING_COUNT = 5;
export const ICON = {
  url: `img/pin.svg`,
  size: [30, 30]
};
export const LEAFLET_OPTION = {
  center: [52.38333, 4.9],
  zoom: 12
};
export const TILE_LAYER = {
  img: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  attributions: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
};
