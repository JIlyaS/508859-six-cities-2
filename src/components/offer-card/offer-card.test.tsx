import * as React from 'react';
import * as Enzyme from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as Adapter from 'enzyme-adapter-react-16';
import {shape} from 'prop-types';
import {BrowserRouter} from 'react-router-dom';

import OfferCard from './offer-card';
import {OfferCardMock} from '../../constants';

Enzyme.configure({adapter: new Adapter()});

const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
};

const createContext = () => ({
  context: {router},
  childContextTypes: {router: shape({})},
});

it(`OfferCard correctly renders after relaunch`, () => {
  const tree = Enzyme.shallow(<OfferCard
    offer={OfferCardMock.OFFER}
    offerId={OfferCardMock.OFFER_ID}
    onActiveOfferMouseEnter={() => {}}
    onDeactiveOfferMouseLeave={() => {}}
    onFavoriteOfferClick={() => {}}
    classCard={OfferCardMock.CLASS_CARD}
  />, createContext());

  expect(toJSON(tree)).toMatchSnapshot();
});
