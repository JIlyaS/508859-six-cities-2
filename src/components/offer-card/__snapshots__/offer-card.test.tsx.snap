// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard correctly renders after relaunch 1`] = `
<article
  className="card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="room.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className=" place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          0
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button false button"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/0"
      />
    </h2>
    <p
      className="place-card__type"
    >
      Room
    </p>
  </div>
</article>
`;
